src/components/UploadNoStyle/UploadNoStyle.tsx

```diff
@@ -67,8 +72,18 @@ export const UploadNoStyle = (props: UploadNoStyleProps) => {
         props.customRequest
           ? async req => {
               setIsUploading(true)
+              setPercent(0)
               try {
-                const val = await props.customRequest?.(req.file)
+                const val = await props.customRequest?.({
+                  ...req,
+                  onProgress: e => {
+                    req.onProgress?.(e)
+                    if (e) {
+                      // Bit of a hack
+                      setPercent(e as unknown as number)
```

Comment:
the typing for this object looks like `e: {percent: number}`, is it just wrong?

Reply:
unsure, but had to really hack it to get it to work. Antd stuff is just not very friendly unfortunately. Here, they are bleeding another libs interface too. I think the goal should be for us to just rebuild the upload functionality (as we already compose our own custom upload event anyway)